deku-c/deku-c-toolkit/src/contract.ts

```diff
@@ -82,48 +83,97 @@ export class Contract {
   }
 
   /**
-   * Invoke a deku-c smart contrat
-   * @param parameter the parameter of the contract
+   * Invoke a deku-c smart contrat with a tunac-provided expression
+   * @param parameter the parameter of the contract as provided by tunac
    * @returns the hash of the operation
    */
-  async invoke(parameter: any): Promise<string> {
+  async invokeRaw(parameter: any): Promise<string> {
     const invoke = {
       operation: JSON.stringify({
         address: this.address,
-        argument: parameter
+        argument: parameter,
       }),
-      tickets: []
+      tickets: [],
     };
     const hash = await this.deku.submitVmOperation(invoke);
     return hash;
   }
 
+  async invoke(expression: string, address: string): Promise<string> {
+    const parameter = { expression, address };
+    const invoke = await compileExpression(this.deku.dekuRpc, parameter);
+    const hash = await this.deku.submitVmOperation(invoke);
+    return hash;
+  }
+
   /**
-   * Returns the state of the contract
-   * Parses it to a readable javascript object
-   * @returns javascript object
+   * Compiles a Ligo argument and invokes a deku-c smart contract
+   * @param parameter the parameter of the contract, in Ligo // FIXME lang
+   * @returns the hash of the operation
    */
-  async getState(): Promise<any | null> {
-    const state = await this.getRawState();
-    if (state === null) return null;
-    return parseContractState(state);
+  async invokeLigo(
+    code: string,
+    expression: string,
+    ligoRpc: string,
+    dekuRpc: string
```

Comment:
Why do you need the dekuRpc ? Can't it be passed to the contract via it's constructor ? And be a attribute of the contract ?

Reply:
In needed it to compile the Michelson code before posting the operation:

```js
const result = await fetch(ligoRpc + "/api/v1/ligo/expression", options);
const { expression } = await result.json();
return compileExpression(dekuRpc, { expression, address });
```

This was before being able to access dekuRpc into the toolkit, but I guess we don't need to pass it around now.